src/target/riscv/riscv.c

```diff
@@ -2453,87 +2464,46 @@ static int riscv_deassert_reset(struct target *target)
 	return tt->deassert_reset(target);
 }
 
-/* state must be riscv_reg_t state[RISCV_MAX_HWBPS] = {0}; */
-static int disable_triggers(struct target *target, riscv_reg_t *state)
+/* state must be riscv_reg_t state[r->trigger_count] = {false}; */
```

Comment:
Have you meant this?

```suggestion
/* wp_is_set array must have at least "r->trigger_count" items. */
```
Also I'd prefer to zero-initialize the array inside disable_watchpoints(), as opposed to relying on the caller to do it.  This way the likelihood of coding mistakes would be reduced a little (defensive programming).

```
static int disable_watchpoints(struct target *target, bool *wp_is_set)
{
	LOG_TARGET_DEBUG(target, "Disabling triggers.");

	/* Initialize all wp_is_set items to false */
	for (unsigned int k; k < r->trigger_count; k++)
		wp_is_set[k] = false;

	// ...
```

Reply:
Addressed.